File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()

        for idx in range(len(nums)):
            if ((idx > 0) and (nums[idx] == nums[idx -1])):
                #do not count it
                continue
            
            l, r = idx+1, len(nums)-1
            while l < r:
                threeSome = nums[idx] + nums[l] + nums[r]
                if threeSome > 0:
                    r -= 1
                elif threeSome < 0:
                    l += 1
                else:
                    res.append([nums[idx],nums[l], nums[r]])
                    l += 1
                    while nums[l] == nums[l-1] and l < r:
                        l += 1
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
#include <unordered_map>
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        vector<vector<int>> freq(nums.size()+1);

        for (int n : nums){
            count[n] += 1;
        }
        for (const auto& entry : count){
            freq[entry.second].push_back(entry.first);
        }

        vector<int> res;
        for(int i = freq.size() -1; i >0; --i){
            for (int n : freq[i]){
                res.push_back(n);
                if (res.size() == k){
                    return res;
                }
            }
        }

        return res;
    }
};
/*
Input: nums = [1,2,2,3,3,3], k = 2

{
    1:1,
    
}

*/